src/package-linker.js

```diff
@@ -459,16 +477,16 @@ export default class PackageLinker {
           if (!(peerPkgRef && peerPkgRef.patterns)) {
             continue;
           }
-          const levelDistance = ref.level - peerPkgRef.level;
-          if (levelDistance >= 0 && levelDistance < resolvedLevelDistance) {
+          const levelDistance = getLevelDistance(peerPkgRef);
+          if (isFinite(levelDistance) && levelDistance < resolvedLevelDistance) {
```

Comment:
Is it possible to not set `minDistance` to Infinity and just check for existence now?

Reply:
I wanted that too but then I'd need to add special casing into the `getLevelDistance` code above due to the `distance >= 0 && distance < minDistance` checks. Is it too confusing right now? :(

Reply:
Ah I see, hmm can't think of an alternative atm. Thanks for explaining. :)